### ISSUE TITLE
Have you tried reinforcement MnemonicReader？

### ISSUE THREAD

**Issue by robbine** [obryanlouis/qa#8]
hi obryanlouis,
I read the paper " reinforcement mnemonic reader". It says " To further stabilize training and prevent the model from overwriting its earlier training, we integrate the maximum- likelihood estimation with the reinforcement learning by using a linear interpolation"
Therefore I tried to add ctc_loss to further improve f1 score, however the ctc_loss became INF. here is my code forked from yours: https://github.com/robbine/qa/blob/master/model/reinforcement_fusion_net.py
https://github.com/robbine/qa/blob/master/model/connectionist_network_pointer.py
By the way, I use tfrecord format instead of zipped dataset.

**Comment by obryanlouis** [comment]
I don't know if I have a full grasp of everything, but as I understood the paper, they basically calculated the F1 score and used that in interpolation like you are doing with the CTC loss. So it seems similar. 

In the past, I found the TF debugging tools helpful for figuring out why there are Nan's or INF's: https://www.tensorflow.org/programmers_guide/debugger. It can point you to which operation gives bad values first.

I stopped working on this project, but if I was going to continue I might try integrating Elmo weights into the model. It apparently can give a ~3-4% boost. I started with that but couldn't get it to work quite right
